### PR TITLE
Add cohort to user sync

### PR DESCRIPTION
**Pull request by xswoodsx**
## Ticket and context

Ticket: [776](https://dfedigital.atlassian.net/jira/software/projects/CPDEL/boards/89/backlog?selectedIssue=CPDEL-776)

We want to know whether a user is part of the 2020 cohort (NQT + 1) for our analytics data so we need to get this from R&P. We'll also want to distinguish between future cohorts and cut off access to the platform later down the line.

I'll open a separate PR on R&P to send that information. 

### Is there anything weird that code reviewer should know?
Nope, everything is straightforward.

### Review Checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests
- [ ] All forms have an `id` attribute on them

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -46,6 +46,11 @@ def core_induction_programme
     participant_profile&.core_induction_programme
   end
 
+  def cohort
+    return early_career_teacher_profile.cohort if early_career_teacher?
+    return mentor_profile.cohort if mentor?
+  end
```
Comment (kicferk1) [inline app/models/user.rb:52]: `participant_profile&.cohort`?

**Review by kicferk1 — COMMENTED** [review]
Should we add cohorts to our base seed?

File: app/models/user.rb (hunk 1 above)
Comment (xswoodsx) [reply app/models/user.rb:52]: > Should we add cohorts to our base seed?

yeah probably make sense to.

File: spec/models/analytics_data_layer_spec.rb [hunk 2]
```diff
@@ -27,6 +27,12 @@
       expect(data_layer.analytics_data[:cipYear]).to eq("Year #{course_year.position}")
     end
 
+    it "should add the correct cohort year to the analytics data" do
+      user.early_career_teacher_profile.update!(cohort: create(:cohort, start_year: 2020))
+      data_layer.add_user_info(user)
+      expect(data_layer.analytics_data[:cohort]).to eq(user.cohort)
```
[the remark this answers is not in the document]
Comment (xswoodsx) [reply spec/models/analytics_data_layer_spec.rb:33]: Good spot, yeah should probably just be there start year. We return the core induction programme object in line 34 on the analytics data later. Presumably that should just be the cip name?

**Comment by xswoodsx** [comment]
> Should we add cohorts to our base seed?

yeah probably best to.

File: spec/models/analytics_data_layer_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (FriedSock) [reply spec/models/analytics_data_layer_spec.rb:33]: Yeah I think we should use this `add_cip_info` method that is already being used in `add_year_info`, good spot

File: app/models/analytics_data_layer.rb [hunk 3]
```diff
@@ -31,7 +31,12 @@ def add(data = {})
   def add_user_info(user)
     @analytics_data[:userType] = user.user_description if user
     @analytics_data[:userRegisterAndPartnerId] = user.register_and_partner_id if user
-    @analytics_data[:userCoreInductionProgramme] = user.core_induction_programme if user
+    @analytics_data[:userCoreInductionProgramme] = add_cip_info(user.core_induction_programme) if user
+    @analytics_data[:cohortStartYear] = add_cohort_start_year(user) if user&.participant?
+  end
+
+  def add_cohort_start_year(user)
+    user.cohort&.start_year
```
Comment (kicferk1) [inline app/models/analytics_data_layer.rb:39]: That does not look right 😂 

`add_xxx` are methods that do `@analytics_data[:yyy] = zzz`

I would do
```
    # in add_user_info
    @analytics_data[:cohortStartYear] = user.cohort.start_year if user&.participant?
```
or
```
    # in add_user_info
    add_cohort_start_year(user) if user&.participant?

  def add_cohort_start_year(user)
    @analytics_data[:cohortStartYear] = user.cohort.start_year if user&.participant?
  end
```

Comment (xswoodsx) [reply app/models/analytics_data_layer.rb:39]: My bad, yep, i'll go with the former.

**Review by FriedSock — APPROVED** [review]

**Pull request merged**